src/pkg/cli/configSet_test.go

```diff
@@ -12,10 +12,22 @@ import (
 func TestConfigSet(t *testing.T) {
 	ctx := context.Background()
 	provider := MustHaveProjectNamePutConfigProvider{}
-	err := ConfigSet(ctx, "test", provider, "test_name", "test_value")
-	if err != nil {
-		t.Fatalf("ConfigSet() error = %v", err)
-	}
+
+	t.Run("expect no error", func(t *testing.T) {
+		err := ConfigSet(ctx, "test", provider, "test_name", "test_value")
+		if err != nil {
+			t.Fatalf("ConfigSet() error = %v", err)
+		}
+	})
+
+	t.Run("expect error on DryRun", func(t *testing.T) {
+		DoDryRun = true
+		defer func() { DoDryRun = false }()
```

Comment:
```suggestion
Use `t.Cleanup` in tests instead of `defer` so that we can ensure it will run even if `t.Fatal` is called.

```suggestion
		t.Cleanup(func() { DoDryRun = false })
```
```